engineering/standards/frontend.md

```diff
@@ -326,6 +326,104 @@ TypeScript types (including interfaces) which describe API requests and response
 
 ## Svelte
 
+### Do not dispatch events from components
+
+⚠️ **Note:** This is a new standard and our codebase is not yet in compliance with it. _New_ code should adhere to the standard, and refactoring should take this standard into account when feasible.
+
+Svelte has an [events](https://svelte.dev/docs#template-syntax-component-directives-on-eventname) system whereby components can dispatch events to imperatively pass messages and data to their parents.
+
+- ✅ It's okay to listen to Svelte events _on DOM nodes_. For example:
+
+    ```svelte
+    <button on:click={handleClick}>
+    ```
+
+- ❌ It's not okay to use events on our components
+
+    ```svelte
+    <Button on:click={handleClick}>
+    ```
+
+- ✅ **Our components should use props instead of events**
+
+    ```svelte
+    <Button onClick={handleClick}>
+    ```
+
+- When consuming third party components that use events, it's okay to listen to those events directly. (No need to wrap it in a component that complies with our code standards.)
+
+- For example, in a `Child.svelte` component:
+
+    - ❌ Events are dispatched as follows. Don't do this.
+
+        ```svelte
+        <script lang="ts">
+          import { createEventDispatcher } from 'svelte';
+
+          const dispatch = createEventDispatcher<{ valueChange: number }>();
+
+          dispatch('valueChange', 42);
+        </script>
+        ```
+
+    - ✅ Use props instead.
+
+        ```svelte
+        <script lang="ts">
+          export onValueChange: (value: number) => void;
+
+          onValueChange(42);
+        </script>
+        ```
+
+- Then, in the consuming (parent) component:
+
+    - ❌ Events would be handled as follows:
+
+        ```svelte
+        <script lang="ts">
+          import Child from `./Child.svelte`;
+
+          function handleValueChange({ detail: value }: CustomEvent<number>) {
+                                  // ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Yuck!
+            console.log(`Child reports that its value is: ${value}`)
+          }
+        </script>
+
+        <Child on:valueChange={handleValueChange}/>
+        ```
+
+    - ✅ With props:
+
+        ```svelte
+        <script lang="ts">
+          import Child from `./Child.svelte`;
+
+          function handleValueChange(value: number) {
+                                  // ^^^^^^^^^^^^^ Nicer!
+            console.log(`Child reports that its value is: ${value}`)
+          }
+        </script>
+
+        <Child onValueChange={handleValueChange}/>
+        ```
+
+- Rationale
+
+    | | Events | Props |
+    | --  | -- | -- |
+    | Synchronicity | ❌ Events are always synchronous. There's no way to await the `dispatch` call. | ✅ Props give us the ability to define asynchronous callbacks and then await them within the component. We do this within `SpinnerButton` for example. |
```

Comment:
This rationale confuses closure callbacks with events. Both are very different and have different purposes.

When a component requires the ability to know and control the state of the parent, we need to use closure callbacks, which can be sync or async. Closure callbacks have to be specified as props, since they are properties of the component.

Events however need to be used in cases where the component does not have to bother about how the consumers utilize them.